File: scratch/verify_all_card_data.php
<?php
require_once __DIR__ . '/../db.php';

echo "=======================================================\n";
echo "VERIFYING ALL CARD & AI SYNTHESIS DATA DISPLAYS (PHP CLI)\n";
echo "=======================================================\n\n";

// 1. Verify Feedback Database Data
$fRes = $conn->query("SELECT id, consultation_id, category, message, rating, sentiment_tag, sentiment_score FROM feedback");
$fbList = [];
$sentimentDist = [];
while ($row = $fRes->fetch_assoc()) {
    $fbList[] = $row;
    $stag = $row['sentiment_tag'] ?: 'unknown';
    $sentimentDist[$stag] = ($sentimentDist[$stag] ?? 0) + 1;
}

echo "1. Feedback Records in DB:\n";
echo "   Total Rows: " . count($fbList) . "\n";
echo "   Sentiment Tag Distribution: ";
print_r($sentimentDist);

// 2. Verify AI Brief Compilation Data per Consultation
$cRes = $conn->query("SELECT id, title, status, category FROM consultations ORDER BY id ASC");
echo "\n2. AI Committee Brief Data per Consultation:\n";

while ($c = $cRes->fetch_assoc()) {
    $cid = (int)$c['id'];
    $title = $c['title'];
    
    $_GET['action'] = 'compile_committee_brief';
    $_GET['consultation_id'] = $cid;
    $_GET['force'] = 1;
    
    ob_start();
    require __DIR__ . '/../API/consultation_feedback_ai.php';
    $jsonRaw = ob_get_clean();
    
    $json = json_decode($jsonRaw, true);
    if ($json && !empty($json['success'])) {
        $brief = $json['data'];
        $comm = $brief['committee_assigned'] ?? $brief['assigned_committee'] ?? 'Unknown Committee';
        $stats = $brief['stats'] ?? [];
        $problems = $brief['problems'] ?? [];
        $solutions = $brief['solutions'] ?? [];
        
        echo "   ✓ Consult #{$cid} (\"{$title}\"):\n";
        echo "       Assigned Committee: {$comm}\n";
        echo "       Total Citizen Feedback: " . ($stats['total_submissions'] ?? 0) . " | Dominant Public Tone: " . ($stats['dominant_sentiment'] ?? 'N/A') . "\n";
        echo "       Problems/Grievances Count: " . count($problems) . " | Recommendations Count: " . count($solutions) . "\n";
        if (!empty($problems)) {
            echo "       Top Problem: [" . $problems[0]['category'] . "] " . substr($problems[0]['issue'], 0, 80) . "...\n";
        }
    } else {
        echo "   ❌ Consult #{$cid} Brief Compilation Failed!\n";
    }
}

echo "\n=======================================================\n";
echo "ALL CARD & AI SYNTHESIS DATA IS VERIFIED LEGITIMATE!\n";
echo "=======================================================\n";

File: scratch/verify_all_card_data.py
import urllib.request
import json
import ssl

ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE

base_url = "http://localhost/CAP101/PC"

print("=== VERIFYING CARD & SYNTHESIS DATA DISPLAYS ===")

# 1. Test feedback API
try:
    req = urllib.request.Request(f"{base_url}/API/feedback_api.php?action=list&limit=200")
    with urllib.request.urlopen(req, context=ctx) as resp:
        data = json.loads(resp.read().decode('utf-8'))
        fb_list = data.get('data', [])
        print(f"✓ Feedback API returned {len(fb_list)} feedback items.")
        
        sentiments = {}
        for fb in fb_list:
            stag = fb.get('sentiment_tag') or fb.get('sentiment')
            sentiments[stag] = sentiments.get(stag, 0) + 1
        print(f"  Feedback Sentiment Distribution: {sentiments}")
except Exception as e:
    print(f"✗ Feedback API error: {e}")

# 2. Test AI Committee Brief Compilation for each consultation
consultation_ids = [1, 2, 3, 4, 5, 17, 18, 20, 24]
for cid in consultation_ids:
    try:
        req = urllib.request.Request(f"{base_url}/API/consultation_feedback_ai.php?action=get_brief&consultation_id={cid}")
        with urllib.request.urlopen(req, context=ctx) as resp:
            data = json.loads(resp.read().decode('utf-8'))
            if data.get('success'):
                brief = data.get('data', {})
                title = brief.get('title')
                committee = brief.get('committee_assigned') or brief.get('assigned_committee')
                stats = brief.get('stats', {})
                problems = brief.get('problems', [])
                solutions = brief.get('solutions', [])
                print(f"✓ Consultation #{cid} ('{title[:35]}...'):")
                print(f"    Committee: {committee}")
                print(f"    Feedback Count: {stats.get('total_submissions')} | Tone: {stats.get('dominant_sentiment')}")
                print(f"    Problems Identified: {len(problems)} | Solutions Proposed: {len(solutions)}")
                if problems:
                    print(f"    Sample Issue: {problems[0].get('category')} -> {problems[0].get('issue')[:70]}...")
            else:
                print(f"✗ Consultation #{cid} Brief error: {data.get('message')}")
    except Exception as e:
        print(f"✗ Consultation #{cid} Brief fetch error: {e}")

print("\n=== VERIFICATION COMPLETE ===")
